iFixit/checkout-ui-extensions
# Add EU google tag custom pixel to git

Closes https://github.com/iFixit/checkout-ui-extensions/issues/38

Similar to https://github.com/iFixit/checkout-ui-extensions/pull/65 this adds the custom pixel for the EU Google Tag Manager tag to git for a custom pixel. Shopify has a [tutorial](https://help.shopify.com/en/manual/promoting-marketing/pixels/custom-pixels/gtm-tutorial) that shows GTM being initialized the same way we're doing here.

Copies the [shopify-v2 implementation](https://github.com/iFixit/shopify-v2/blob/4e1c1341fb89d66c697f934b6b9b127e0be1ae47/src/layout/checkout.liquid#L4-L15). Shopify doesn't seem to offer a way to add the [noscript integration](https://github.com/iFixit/shopify-v2/blob/4e1c1341fb89d66c697f934b6b9b127e0be1ae47/src/layout/checkout.liquid#L437-L444), but that's probably ok.

I imagine that to "deploy" this, we'll need to eventually copy this snippet to a new custom pixel on each store and also make sure we use the correct GTM_ID (can be found in theme settings for each store). It would be nifty to generate a JS file for each store using a template populated with the GTM_ID from a `.env.local` file. But maybe that wouldn't really save any time, because we'd still have to copy each JS file to Shopify admin manually.

## QA

QA will be tricky until we really go live and see data coming into the dashboard. We can use [Google Tag Assistant Legacy Chrome extension](https://chrome.google.com/webstore/detail/tag-assistant-legacy-by-g/kejbdjndbnbjgmefkgdddjlbokphdefk) to test custom pixels [according](https://help.shopify.com/en/manual/promoting-marketing/pixels/custom-pixels/gtm-tutorial) to Shopify.
